src/plugins/guided_onboarding/public/services/api.ts

```diff
@@ -55,7 +55,13 @@ export class ApiService implements GuidedOnboardingApi {
         })
         .catch((error) => {
           this.isPluginStateLoading = false;
-          observer.error(error);
+          // if the request fails, we initialize the state with error
+          observer.next({ status: 'error', isActivePeriod: false });
+          this.pluginState$.next({
```

Comment:
Are there any corresponding UI changes needed here, i.e., to show the toast notification if the status === error?

Reply:
Yes, I opened an [issue](https://github.com/elastic/kibana/issues/146437) to follow up on that. This work will probably be done for 8.7 though.